graphql-ballerina/build.gradle

```diff
@@ -159,23 +158,29 @@ test {
 
 task ballerinaBuild {
     inputs.dir file(project.projectDir)
-    dependsOn(":graphql-native:build")
-    dependsOn(unpackJballerinaTools)
-    dependsOn(unpackStdLibs)
-    dependsOn(copyStdlibs)
     dependsOn(updateTomlFile)
-    dependsOn(test)
+    dependsOn(initializeVariables)
     finalizedBy(revertTomlFile)
 
+    def testParams = "--skip-tests"
+    gradle.taskGraph.whenReady { graph ->
+        if (graph.hasTask(':graphql-ballerina:build')) {
+            ballerinaTest.enabled = false
+        }
+        if (graph.hasTask(':graphql-ballerina:test')) {
+            testParams = "--code-coverage"
+        }
+    }
+
     doLast {
         // Build and populate caches
         exec {
             workingDir project.projectDir
             environment "JAVA_OPTS", "-DBALLERINA_DEV_COMPILE_BALLERINA_ORG=true"
             if (Os.isFamily(Os.FAMILY_WINDOWS)) {
-                commandLine 'cmd', '/c', "$distributionBinPath/bal.bat build --skip-tests -c"
```

Comment:
There is another task to run tests AFAIK. Do we need to update this part in that case?

Reply:
If we skip tests here, the build will run the test task first, and then run the build step after that. When we skip tests in build step, the code-coverage files will be lost. To handle that, we only run the test task if the build task is not present. (In line no. 166).

This way, we can preserve the code coverage files, and also keep the build as it is.